### ISSUE TITLE
"Route to Mode" still borked

### ISSUE THREAD

**Issue by saamohod** [LoneGazebo/Community-Patch-DLL#9887]
_1. Mod Version (X.Y.Z). Current Version: 3.6.1_

The issue that first appeared in 3.6 still persists. When ordered to build a railroad by "Route to Mode" button, the worker doesn't consider an existing road to build a railroad over it (the way it has always been before).

Example:
The worker that sits on a road tile 5 o'clock from Marrakech is ordered to build a railroad to Rabat (to the east of Marrakech) via "Route to Mode". Instead of sticking to the road, the worker steps away onto the roadless hill tile with the gold resource :

![20230527094112_1](https://github.com/LoneGazebo/Community-Patch-DLL/assets/16341598/8f10cf5d-c9bf-4ad7-b18c-eae9d80200d1)
![20230527094129_1](https://github.com/LoneGazebo/Community-Patch-DLL/assets/16341598/a5a16287-0177-4488-a6bc-cf146188bed6)

**Comment by saamohod** [comment]
> It is picking the shortest route to connect to.. but unsure of how the behavior should be treated as.

Yes it seems so. However it wasn't this way all the time I played this mod for years. It used to utilize certain common sense and use an existing road unless the existing road is too far away. The change happened in 3.6. Now the it doesn't seem to see the old road at all, before it took it into account.

**Comment by KungCheops** [comment]
I'll take a look
